pywhy_graphs/algorithms/generic.py

```diff
@@ -333,3 +334,207 @@ def _single_shortest_path_early_stop(G, firstlevel, paths, cutoff, join, valid_p
+    directed_children = set(G.sub_directed_graph().successors(node))
+
+    out = bidirected_children.union(directed_children)
+
+    return out
+
+
+def _is_collider(G, node):
+    """Checks if the given node is a collider or not.
+
+    Parameters
+    ----------
+    G : graph
+        The graph.
+    node : node
+        The node to be checked.
+
+    Returns
+    -------
+    iscollider : bool
+        Bool is set true if the node is a collider, false otherwise.
+    """
+    parents = _find_directed_parents(G, node)
+
+    if len(parents) > 1:
+        return True
+    return False
+
+
+def _recursive_path(G, node_x, node_y, L, S, visited, xyancestors, cur_node):
+    """Recursively explores a graph to find a path.
+
+       Finds path that are compliant with the inducing path requirements.
+
+    Parameters
+    ----------
+    G : graph
+        The graph.
+    node_x : node
+        The source node.
+    node_y : node
+        The destination node
+    L : set
+        Set containing all the non-colliders.
+    S : set
+        Set containing all the colliders.
+    visited : set
+        Set containing all the nodes already visited.
+    xyancestors : set
+        Set containing the ancestors of X and Y.
+    cur_node : node
+        The current node.
+
+    Returns
+    -------
+    path : Tuple[bool, path]
+        A tuple containing a bool and a path which is empty if the bool is false.
+    """
+    path_exists = False
+    path = []
+    visited.add(cur_node)
+    children = _find_directed_children(G, cur_node)
+
+    if cur_node is node_y:
+        return (True, [node_y])
+
+    for elem in children:
+        if elem in visited:
+            continue
+        else:
+            if (
+                _is_collider(G, elem)
+                and (elem not in S)
+                and (elem not in xyancestors)
+                and (elem is not node_y)
+            ):
+                continue
+            elif not _is_collider(G, elem) and (elem not in L) and (elem is not node_y):
+                continue
+            path_exists, temp_path = _recursive_path(
+                G, node_x, node_y, L, S, visited, xyancestors, elem
+            )
+            if path_exists:
+                path.append(cur_node)
+                path.extend(temp_path)
+                break
+    return (path_exists, path)
+
+
+def inducing_path(G, node_x, node_y, L=None, S=None):
+    """Checks if an inducing path exists between node_x and node_y and if it does returns it.
+
+    Parameters
+    ----------
+    G : Graph
+        The graph.
+    node_x : node
+        The source node.
+    node_y : node
+        The destination node.
+    L : set
+        The set containing every non-collider on the path. Defaults to an empty set.
+    S:  set
+        Contains all colliders on the path. Defaults to an empty set.
+
+
+    Returns
+    -------
+    path : Tuple[bool, path]
+        A tuple containing a bool and a path if the bool is true.
+    """
```

Comment:
Add a `Notes` section describing i) what is the intuition of an inducing path and ii) describing how the hyper parameters ``L`` and ``S`` are used.